### PR TITLE
[Feature] Support DBNet, PANet and PSENet for SDK, with GPU aided post-processing

### PR DESCRIPTION
**Pull request by lzhangzz**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

- Add post-processing for PANet & PSENet.
- Add full CUDA support for DBNet post-processing and partial CUDA support for PANet and PSENet post-processing.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by AllentDan — COMMENTED** [review]
Hi, @lzhangzz. It failed when building SDK in my Linux.
```
[ 61%] Building CXX object csrc/preprocess/cuda/CMakeFiles/mmdeploy_cuda_transform_impl.dir/default_format_bundle_impl.cpp.o
CMakeFiles/Makefile2:1280: recipe for target 'csrc/codebase/mmocr/cuda/CMakeFiles/mmdeploy_mmocr_cuda_impl.dir/all' failed
make[1]: *** [csrc/codebase/mmocr/cuda/CMakeFiles/mmdeploy_mmocr_cuda_impl.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

File: csrc/codebase/mmocr/cuda/connected_component.cu [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+  size_t size = height * width;
+  if (size > capacity_) {
+    if (!capacity_) {
+      capacity_ = size;
+    } else {
+      while (capacity_ < size) {
+        capacity_ *= growth_factor_;
+      }
+    }
+    cudaFree(d_label_);
```
Comment (grimoire) [inline csrc/codebase/mmocr/cuda/connected_component.cu:372]: Do we have a plan to add some kind of allocator module?

**Review by grimoire — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]
LGTM

**Review by grimoire — APPROVED** [review]

File: csrc/codebase/mmocr/cuda/connected_component.cu (hunk 1 above)
Comment (lzhangzz) [reply csrc/codebase/mmocr/cuda/connected_component.cu:372]: We have an device allocator module, but it's not good enough yet.

**Pull request merged**